app/services/order_service.rb

```diff
@@ -32,20 +32,10 @@ def self.set_shipping!(order, fulfillment_type:, shipping: {})
   end
 
   def self.approve!(order, by: nil)
-    order.approve! do
-      charge = PaymentService.capture_charge(order.external_charge_id)
-      transaction = {
-        external_id: charge.id,
-        source_id: charge.source,
-        destination_id: charge.destination,
-        amount_cents: charge.amount,
-        failure_code: charge.failure_code,
-        failure_message: charge.failure_message,
-        transaction_type: charge.transaction_type,
-        status: Transaction::SUCCESS
-      }
-      TransactionService.create!(order, transaction)
+    transaction = order.approve! do
+      transaction = PaymentService.capture_charge(order.external_charge_id)
```

Comment:
Do you need to assign `transaction` here again?

Reply:
nope :) 👀